src/telegram_to_engagement_db/telegram_group_to_engagement_db.py

```diff
@@ -0,0 +1,240 @@
+        log.debug(f"message already in engagement database")
+        return
+
+    log.debug(f"Adding message to engagement database")
+    engagement_db.set_message(
+        telegram_message,
+        HistoryEntryOrigin(origin_name="Telegram Group -> Database Sync", details=message_origin_details)
+    )
+
+
+async def sync_messages_from_groups_to_engagement_db(telegram_group_source, telegram,
+                                                     engagement_db, uuid_table, cache_path):
+    """
+    :param telegram_group_source: Telegram sources to sync to the engagement database.
+    :type telegram_group_source: List of src.telegram_to_engagement_db.configuration.TelegramGroupSource
+    :param telegram: Instance of telegram app to use to download the group messages from.
+    :type telegram: telethon.client.telegramclient.TelegramClient
+    :param uuid_table: UUID table to use to re-identify the URNs so we can set the channel operator.
+    :type uuid_table: id_infrastructure.firestore_uuid_table.FirestoreUuidTable
+    :param cache_path: Path to a directory to use to cache results needed for incremental operation.
+                       If None, runs in non-incremental mode.
+    :type cache_path: str | None
+    """
+    if cache_path is None:
+        cache = None
+    else:
+        log.info(f"Initialising TelegramSyncCache at '{cache_path}/telegram_group_to_engagement_db'")
+        cache = TelegramGroupSyncCache(f"{cache_path}/telegram_group_to_engagement_db")
+
+    for dataset in telegram_group_source.datasets:
+        log.info(f"Fetching messages for {dataset.engagement_db_dataset}...")
+        dataset_start_date = datetime.fromisoformat(dataset.search.start_date)
+        dataset_end_date = datetime.fromisoformat(dataset.search.end_date)
+
+        for group_id in dataset.search.group_ids:
+            group_cache_entry_name = f"{dataset.engagement_db_dataset}_{group_id}"
+            group_min_id = None if cache is None else cache.get_latest_group_message_id(group_cache_entry_name)
+
+            # Fetch group messages sent before the dataset_end_date and/or contain message.id greater than min_id in
+            # cache if available.
+            group_messages = await _fetch_message_from_group(telegram, group_id, dataset_end_date, group_min_id)
+
+            broad_cast_admin_messages = 0
+            dataset_group_latest_seen_message_id = None
+            async for telegram_message in group_messages:
+                if _is_avf_message(telegram_message):
+                    broad_cast_admin_messages += 1
+                    continue
+
+                # Filter messages sent between this dataset start and end_time.
+                if dataset_start_date <= telegram_message.date < dataset_end_date:
+                    message_origin_details = {"message_id": telegram_message.id, "group_id":
+                                              telegram_message.peer_id.channel_id}
```

Comment:
I think this should include the full message object, including its text and timestamp, not just the id